### ISSUE TITLE
⌨ Navigation: Close menu button is not accessible using keyboard

### ISSUE THREAD

**Issue by MaddyDaigle** [GCTC-NTGC/gc-digital-talent#12887]
## ⌨ Accessibility Issue

When viewing the application on a smaller screen where the navigation is displayed on the bottom right, keyboard users can open the navigation menu but cannot close it using the "close menu" button by tabbing through the options. The only way to close the menu is by selecting the house icon, which navigates them back to the main page, disrupting their flow and causing confusion.

### WCAG 2.1 Criteria

WCAG 2.1.1 Keyboard – All functionality must be accessible via keyboard, including closing the navigation menu.

WCAG 2.4.3 Focus Order – Focus order should be logical and intuitive, ensuring that the "close menu" button is included in the tab sequence.

## 🦋 Expected Behaviour

Keyboard users should be able to tab to the "close menu" button and activate it to close the navigation menu without being redirected to the main page. The focus order should be logical, ensuring that the user can navigate through all elements within the menu before continuing to other parts of the page.

## 🕵️ Details

- **Operating System**: Windows 10
- **Browser**: Edge
- **Assistive Technology**: Keyboard-only

## 📋 Steps to Reproduce

1. Reduce the size of the browser window (or increase the font size) so that the menu appears on the bottom right.
2. Tab to the menu using a keyboard and open it.
3. Attempt to tab to the "close menu" button.

## 📸 Screenshot

![Image](https://github.com/user-attachments/assets/a6b50880-4eee-457d-92fe-f22618e0e7f4)

**Comment by esizer** [comment]
This is sadly because the mobile menu has a focus trap on it so you cannot shift focus outside and onto elements hidden by it.

The obvious solution is to not display the button outside and put one inside the menu but that would require kicking back to design.

**Comment by MaddyDaigle** [comment]
@esizer Yea, this is a biggy and fails a WCAG level A criteria. I'd definitely prioritize this fix.

**Comment by gobyrne** [comment]
> This is sadly because the mobile menu has a focus trap on it so you cannot shift focus outside and onto elements hidden by it.
> 
> The obvious solution is to not display the button outside and put one inside the menu but that would require kicking back to design.

@esizer What's stopping us from including the "Close menu" button inside the focus trap? Even if it ends up being a separate semantic element from the "Open menu" item, that shouldn't be a big deal.

I feel like we can put the close menu button inside and it will work as designed, no?
